src/System.Numerics.Vectors/tests/Performance/Vector2/DistanceSquared.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Xunit.Performance;
+
+namespace System.Numerics.Tests
+{
+    public static partial class Perf_Vector2
+    {
+        [Benchmark(InnerIterationCount = VectorTests.DefaultInnerIterationsCount)]
+        public static void DistanceSquaredBenchmark()
+        {
+            const float expectedResult = 134217728.0f;
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                float result;
+
+                using (iteration.StartMeasurement())
+                {
+                    result = DistanceSquaredTest();
+                }
+
+                VectorTests.AssertEqual(expectedResult, result);
+            }
+        }
+
+        public static float DistanceSquaredTest()
+        {
+            var result = 0.0f;
+            var valueX = new Vector2(-1.0f, 1.0f);
+            var valueY = new Vector2(1.0f, -1.0f);
+
+            for (var iteration = 0; iteration < Benchmark.InnerIterationCount; iteration++)
+            {
+                valueX += VectorTests.Vector2Delta;
```

Comment:
It should be fairly minimal, comparatively. Ideally we could remove this, but the JIT can then potentially optimize away the call if the inputs aren't changed (not sure if we do that today, but it is a potential thing that can happen).

Just looking at Vector4, the duration of the AddFunction is `168.505` (is that milliseconds?) and the duration of the functions in question are as follows:
* DistanceSquared: 280.863 (takes 66.6% more time)
* DotTest: 252.822 (takes 50.0% more time
* GetHashCode: 665.347 (takes 294.8% more time)
* LengthSquared: 224.386 (takes 33.1% more time)

**Note:** This doesn't take into account the overhead of the `for` loop (which is shared between both), so it isn't a precise comparison.

The `add` operation is also intrinsic and is essentially as small as it can get already. The assembly it produces is essentially:
* Ensure parameters are in the XMM register
* Call `addps`

So it likely can't be optimized much more (other than maybe reduce the number of register to memory copies or vice-versa).

Any further improvements made to it should end up impacting essentially all downstream consumers, including these perf tests.